### ISSUE TITLE
Add a configuration option for default init options

### ISSUE THREAD

**Issue by m-legrand** [astral-sh/uv#10241]
_This is a feature request_

I like `uv init` to create simple ad-hoc projects and often use it with the exact same options.
I would love to be able to drive this via a configuration, such as a `uv.toml` in a "one-off apps" folder.

For consistency with other subcommands it could look like this:

```uv.toml
# uv.toml

[init]
app = true
vcs = "none"
```

**Comment by adriangalilea** [comment]
Yeah I’m so annoyed by the multiple files for no reason whatsoever when `uv init`ing something. :(
